mlx/coverity_services.py

```diff
@@ -259,356 +209,237 @@ def add_filter_rqt(name, req_csv, valid_list, filter_list, allow_regex=False):
-        snapshot_scope = self.client.factory.create('snapshotScopeSpecDataObj')
-
-        snapshot_scope.showOutdatedStreams = False
-        snapshot_scope.compareOutdatedStreams = False
-
-        snapshot_scope.showSelector = 'last()'
-        snapshot_scope.compareSelector = 'last()'
-
-        logging.info('Running Coverity query...')
-        return self.client.service.getMergedDefectsForSnapshotScope(project_id, filter_spec,
-                                                                    page_spec, snapshot_scope)
+        if filters["cid"]:
+            # this should be a idMatcher (columnKey: cid)
+            filter_list = self.handle_attribute_filter(filters["cid"], "CID", None)
+            if filter_list:
+                self.add_new_filters(request_filters, "cid", filter_list, "idMatcher")
+        column_names = [name.lower() for name in column_names]
+        column_keys = set()
+        for column in self.columns:
+            if column["name"].lower() in column_names:
+                column_keys.add(column["columnKey"])
+        if "location" in column_names:
+            column_keys.add("lineNumber")
+            column_keys.add("displayFile")
+        if "comment" in column_names:
+            column_keys.add("lastTriageComment")
+        if "reference" in column_names:
+            column_keys.add("externalReference")
+        column_keys.add("cid")
+        data = {
+            "filters": request_filters,
+            "columns": list(column_keys),
+            "snapshotScope": {
+                "show": {
+                    "scope": "last()",
+                    "includeOutdatedSnapshots": False
+                },
+                "compareTo": {
+                    "scope": "last()",
+                    "includeOutdatedSnapshots": False
+                }
+            }
+        }
+        logging.info("Running Coverity query...")
+        return self.retrieve_issues(data)
 
     def handle_attribute_filter(self, attribute_values, name, *args, **kwargs):
-        """ Applies any filter on an attribute's values.
+        """Applies any filter on an attribute's values.
 
         Args:
             attribute_values (str): A CSV list of attribute values to query.
             name (str): String representation of the attribute.
+
+        Returns:
+            list[str]: The list of valid attributes
         """
-        logging.info('Using %s filter [%s]', name, attribute_values)
-        validated = self.config_service.add_filter_rqt(name, attribute_values, *args, **kwargs)
-        logging.info('Resolves to [%s]', validated)
+        logging.info("Using %s filter [%s]", name, attribute_values)
+        validated, filter_list = self.add_filter_rqt(name, attribute_values, *args, **kwargs)
+        logging.info("Resolves to [%s]", validated)
         if validated:
-            self.filters += ("<%s(%s)> " % (name, validated))
+            self.filters += "<%s(%s)> " % (name, validated)
```

Comment:
I think that `self.filters` is not used

Reply:
Indeed, I will delete it

Reply:
Resolved in 9eaafca4bba9db067eed9977d701f0f41c64c99e